src/software/ai/hl/stp/skill/skill.h

```diff
@@ -31,15 +31,60 @@ class Skill
     /**
      * Returns whether the given robot has finished executing the Skill.
      *
-     * @param robot the robot executing the Skill
+     * @param robot_id the ID of the robot executing the Skill
      *
      * @return true if the robot has finished executing the Skill, false otherwise
      */
-    virtual bool done(const RobotId& robot_id) const = 0;
+    virtual bool done(const RobotId robot_id) const = 0;
 
-    virtual std::string getFSMState(RobotId robot_id) const = 0;
+    /**
+     * Returns whether the given robot has temporarily suspended execution of the Skill.
+     *
+     * IMPORTANT: updatePrimitive should not be called with a robot that has
+     * suspended execution of the Skill. This is because a suspended Skill
+     * will not yield any primitives for the robot to execute.
+     *
+     * @param robot_id the ID of the robot executing the Skill
+     *
+     * @return true if the robot has suspended execution of the Skill, false otherwise
+     */
+    virtual bool suspended(const RobotId robot_id) const = 0;
 
-    virtual void accept(SkillVisitor& visitor) = 0;
+    /**
+     * If the given robot has temporarily suspended execution of the Skill, calling this
+     * method will update the Skill with the given world and try to resume execution
+     * (i.e. leave the suspended state).
+     *
+     * If the Skill is not suspended for the given robot, then calling this method
+     * effectively does nothing.
+     *
+     * @param robot_id the ID of the robot executing the Skill
+     * @param world_ptr the world pointer
+     *
+     * @return true if the Skill is still suspended for the robot after
+     * trying to resume execution, false otherwise
```

Comment:
semantically the opposite makes sense. It seems more intuitive to return `false` if it failed to resume